packages/cli/src/databases/repositories/workflow.repository.ts

```diff
@@ -99,30 +121,186 @@ export class WorkflowRepository extends Repository<WorkflowEntity> {
+		const [workflowsAndFolders, count] = await Promise.all([
+			this.getWorkflowsAndFoldersUnion(workflowIds, options),
+			this.getWorkflowsAndFoldersCount(workflowIds, options),
+		]);
+
+		const { workflows, folders } = await this.fetchExtraData(workflowIds);
+
+		const enrichedWorkflowsAndFolders = this.enrichDataWithExtras(workflowsAndFolders, {
+			workflows,
+			folders,
+		});
+
+		return [enrichedWorkflowsAndFolders, count] as const;
+	}
+
+	private async fetchExtraData(workflowIds: string[]) {
+		const [workflows, folders] = await Promise.all([
+			this.getMany(workflowIds),
+			this.folderRepository.getMany(),
+		]);
+
+		return { workflows, folders };
+	}
+
+	private enrichDataWithExtras(
+		baseData: WorkflowAndFolderUnion[],
+		extraData: {
+			workflows: ListQuery.Workflow.WithSharing[] | ListQuery.Workflow.Plain[];
+			folders: Folder[];
+		},
+	): WorkflowAndFolderUnionFull[] {
+		return baseData.map((item) => {
+			const extraItem =
+				item.resource === 'folder'
+					? extraData.folders.find((folder) => folder.id === item.id)
+					: extraData.workflows.find((workflow) => workflow.id === item.id);
+
+			return extraItem ? { ...item, ...extraItem } : item;
+		});
+	}
+
+	async getMany(workflowIds: string[], options: ListQuery.Options = {}) {
+		if (workflowIds.length === 0) {
+			return [];
+		}
+
+		const query = this.getManyQuery(workflowIds, options);
+
+		const workflows = (await query.getMany()) as
+			| ListQuery.Workflow.Plain[]
+			| ListQuery.Workflow.WithSharing[];
+
+		return workflows;
+	}
+
+	async getManyAndCount(sharedWorkflowIds: string[], options: ListQuery.Options = {}) {
 		if (sharedWorkflowIds.length === 0) {
 			return { workflows: [], count: 0 };
 		}
 
-		const qb = this.createBaseQuery(sharedWorkflowIds);
+		const query = this.getManyQuery(sharedWorkflowIds, options);
+
+		const [workflows, count] = (await query.getManyAndCount()) as [
+			ListQuery.Workflow.Plain[] | ListQuery.Workflow.WithSharing[],
+			number,
+		];
+
+		return { workflows, count };
+	}
+
+	getManyQuery(workflowIds: string[], options: ListQuery.Options = {}) {
+		const qb = this.createBaseQuery(workflowIds);
 
 		this.applyFilters(qb, options.filter);
 		this.applySelect(qb, options.select);
 		this.applyRelations(qb, options.select);
 		this.applySorting(qb, options.sortBy);
 		this.applyPagination(qb, options);
 
-		const [workflows, count] = (await qb.getManyAndCount()) as [
-			ListQuery.Workflow.Plain[] | ListQuery.Workflow.WithSharing[],
-			number,
-		];
-
-		return { workflows, count };
+		return qb;
 	}
 
-	private createBaseQuery(sharedWorkflowIds: string[]): SelectQueryBuilder<WorkflowEntity> {
-		return this.createQueryBuilder('workflow').where('workflow.id IN (:...sharedWorkflowIds)', {
-			sharedWorkflowIds,
+	private createBaseQuery(workflowIds: string[]): SelectQueryBuilder<WorkflowEntity> {
+		return this.createQueryBuilder('workflow').where('workflow.id IN (:...workflowIds)', {
+			/*
```

Comment:
If the user does not have access to any workflow, we still want to do the union, which will return only folders. To prevent the workflows query to fail, we add a empty value.